src/main/java/org/folio/service/manager/input/InputDataManagerImpl.java

```diff
@@ -92,34 +89,35 @@ protected void initBlocking(JsonObject request, Map<String, String> params) {
-          fileDefinitionService.save(fileExportDefinition, tenantId).onSuccess(savedFileExportDefinition -> {
-            initInputDataContext(sourceReader, jobExecutionId);
-            ExportPayload exportPayload = createExportPayload(okapiConnectionParams, savedFileExportDefinition, jobExecutionId);
-            Optional<JsonObject> optionalUser = usersClient.getById(exportRequest.getMetadata().getCreatedByUserId(), okapiConnectionParams);
-            if (optionalUser.isPresent()) {
-              JsonObject user = optionalUser.get();
-              jobExecutionService.prepareJobForExport(jobExecutionId, fileExportDefinition, user, tenantId);
-              exportNextChunk(exportPayload, sourceReader);
-            } else {
-              finalizeExport(exportPayload, ExportResult.failed(ErrorCode.USER_NOT_FOUND));
-            }
-          });
+    fileDefinitionService.getById(exportRequest.getFileDefinitionId(), tenantId).onSuccess(requestFileDefinition ->
+      jobExecutionService.getById(requestFileDefinition.getJobExecutionId(), tenantId).onSuccess(jobExecution -> {
+        String jobExecutionId = jobExecution.getId();
+        if (requestFileDefinition.getStatus().equals(FileDefinition.Status.COMPLETED)) {
+          FileDefinition fileExportDefinition = createExportFileDefinition(exportRequest, requestFileDefinition, jobExecution);
+          SourceReader sourceReader = initSourceReader(requestFileDefinition, getBatchSize());
+          if (sourceReader.hasNext()) {
+            fileDefinitionService.save(fileExportDefinition, tenantId).onSuccess(savedFileExportDefinition -> {
+              initInputDataContext(sourceReader, jobExecutionId);
+              ExportPayload exportPayload = createExportPayload(okapiConnectionParams, savedFileExportDefinition, jobExecutionId);
+              Optional<JsonObject> optionalUser = usersClient.getById(exportRequest.getMetadata().getCreatedByUserId(), okapiConnectionParams);
+              if (optionalUser.isPresent()) {
+                JsonObject user = optionalUser.get();
+                jobExecutionService.prepareJobForExport(jobExecutionId, fileExportDefinition, user, tenantId);
+                exportNextChunk(exportPayload, sourceReader);
+              } else {
+                finalizeExport(exportPayload, ExportResult.failed(ErrorCode.USER_NOT_FOUND));
+              }
+            });
+          } else {
+            fileDefinitionService.save(fileExportDefinition.withStatus(FileDefinition.Status.ERROR), tenantId);
+            jobExecutionService.updateJobStatusById(jobExecutionId, JobExecution.Status.FAIL, tenantId);
+            sourceReader.close();
+          }
         } else {
-          fileDefinitionService.save(fileExportDefinition.withStatus(FileDefinition.Status.ERROR), tenantId);
+          LOGGER.error(String.format("Failed to start export process, file definition status with id %s is not COMPLETED", exportRequest.getFileDefinitionId()));
```

Comment:
If the export is triggered before the file upload is completed, then the job execution need not fail. WE just can throw wrror on the export API and allow them to retry. 
With the current logic, we'll have to upload the entire file again to continue

Reply:
Sorry to add some comments which were not introduced in this PR, but We shouldn't have missed them in earlier stories.
And we need to correct them somewhere. If you think this is more work than expected, please create a bug and link it to this PR. And then we can close this PR

Reply:
@KVupp according to the endpoint contract, we always return 204 status code, so I don't think we are able to do it. However, we are getting job execution before checking the file definition status. It seems that checking file definition status not needed here because we will fail if we will not found job execution, what do you think? Should we remove this check?

Reply:
It's not possible to start export before file has been uploaded, because the job gets created on upload completion. No need to check file definition status, it's enough to check if the job exists